pkg/wireguard/agent/agent.go

```diff
@@ -364,3 +424,94 @@ func loadOrGeneratePrivKey(filePath string) (key wgtypes.Key, err error) {
 
 	return wgtypes.NewKey(bytes)
 }
+
+// OnIPIdentityCacheChange implements ipcache.IPIdentityMappingListener
+func (a *Agent) OnIPIdentityCacheChange(modType ipcache.CacheModification, ipnet net.IPNet, oldHostIP, newHostIP net.IP,
+	oldID *identity.NumericIdentity, newID identity.NumericIdentity, encryptKey uint8, k8sMeta *ipcache.K8sMetadata) {
+
+	// This function is invoked from the IPCache with the
+	// ipcache.IPIdentityCache lock held. We therefore need to be careful when
+	// calling into ipcache.IPIdentityCache from Agent to avoid potential
+	// deadlocks.
+	// We are only interested in IPCache entries where the hostIP is set, i.e.
+	// updates with oldHostIP set for deletion/updates and newHostIP set
+	// for newly created entries.
+	a.Lock()
+	defer a.Unlock()
+
+	// List of peers affected by this IPCache change
+	updatedPeers := []*peerConfig{}
+
+	// Remove IP from allowedIPs if it was deleted or the host IP changed
+	if modType == ipcache.Delete || !newHostIP.Equal(oldHostIP) {
+		oldNodeName := a.nodeNameByNodeIP[oldHostIP.String()]
+		if peer := a.peerByNodeName[oldNodeName]; peer != nil {
+			peer.removedAllowedIP(ipnet)
+			updatedPeers = append(updatedPeers, peer)
+		}
+	}
+
+	// Add IP to allowedIP if it was added or the host IP changed
+	if modType == ipcache.Upsert || newHostIP != nil {
```

Comment:
The same question as with the host IP change above.

Reply:
Removed.